griptape/drivers/observability/base_observability_driver.py

```diff
@@ -0,0 +1,13 @@
+from abc import ABC, abstractmethod
+from typing import Any
+
+
+class BaseObservabilityDriver(ABC):
+    def __enter__(self) -> None:
+        pass
+
+    def __exit__(self, exc_type, exc_value, traceback) -> None:
+        pass
+
+    @abstractmethod
+    def invoke_observable(self, func, instance, args, kwargs, decorator_args, decorator_kwargs) -> Any: ...
```

Comment:
This is quite a few positional arguments. Is this just the nature of the beast?

Reply:
I agree, I'm not really a fan. I could use kwargs or a dataclass (e.g. `ObservableInvocation` or `ObservableCall`) instead. I know those options aren't very common in the framework, but I am a fan of readability. You wanna see how it looks?

Reply:
Yeah let's create an inner data class (still using attrs) similar to how we do [here](https://github.com/griptape-ai/griptape/blob/0f196f5b47e8bb8308b4d7feba7a44f6c78d6db4/griptape/utils/prompt_stack.py#L15).

Reply:
Added `Observable.Call`